resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -675,9 +675,14 @@ private[spark] class ApplicationMaster(args: ApplicationMasterArguments) extends
     val userThread = new Thread {
       override def run() {
         try {
-          mainMethod.invoke(null, userArgs.toArray)
-          finish(FinalApplicationStatus.SUCCEEDED, ApplicationMaster.EXIT_SUCCESS)
-          logDebug("Done running users class")
+          if(mainMethod == null) {
```

Comment:
Can this be `null`?

Reply:
Also, I don't think this can return `null`. It should return no such method error if not found or return the method.

Reply:
Yes, I think you are definitely correct. It cannot be null. @vanzin is right, and the check should instead ensure the `main` being invoked is static. PR is updated, but I may decline as something is wrong with the way I am testing.